arangod/RestServer/DatabaseFeature.h

```diff
@@ -39,6 +39,8 @@
 
 #include <condition_variable>
 #include <mutex>
+#include <memory>
```

Comment:
Here we have two "lists" droppedDatabases and databases.
I think they can store std::unique_ptr instead of raw pointer.
In such case will be smaller changes in .cpp

Reply:
@MBkkt : It's actually quite complicated to change the "list" types from `TRI_vocbase_t*` to `std::unique_ptr<TRI_vocbase_t>`.
It requires changes in lots of places in DatabaseFeature.cpp, but the most probablematic thing is the ownership "switch" when moving a unique_ptr from `_droppedDatabases` to some temporary local variable for further cleanup.
`absl::FlatHashSet` only has iterators that return a const reference to the pointed-to member, so moving the contents of a `unique_ptr` from such set is actually not easily possible. We could make the vocbases in the lists `std::shared_ptr`s, which would solve this problem, and potentially is even better for the future, so we can return an `std::shared_ptr` from a database lookup function instead of a raw pointer.
But this will require us to change hundreds of internal APIs, so I am reluctant to do it now.
If you still see any room for improvement in this PR, please feel free to push the changes directly into the PR.
Thanks!

Reply:
Ok, I understood